README.tmpl.md

```diff
@@ -0,0 +1,206 @@
+{{.warning}}
+# `gospel` [![Build status](https://github.com/kortschak/gospel/workflows/Test/badge.svg)](https://github.com/kortschak/gospel/actions)
+
+The `gospel` program lints Go source files for misspellings in comments,
+strings and embedded files.
+
+It uses hunspell to identify misspellings and makes use of source code
+information to reduce the rate of false positive spelling errors where
+words refer to labels within the source code.
+
+
+## Installation
+
+Beyond the standard Go installation process, you must also have libhunspell
+and its header files on your system. For a debian-based system this is done
+with `sudo apt install libhunspell-dev`.
+
+
+## Work Flow
+
+`gospel` makes use of module information, and so must be run within a module
+and will only consider source files that are referred to by the module.
+
+`gospel` can be used without configuration in many cases, particularly for
+small source trees where the number of potential errors is likely to be
+small. However, its behaviour can be tuned with [configuration files](#configuration-files).
+The initial setup workflow makes use of the [`.words`](#.words) file to
+build a dictionary of candidate words to ignore.
+
+```
+$ gospel -misspellings=.words ./...
+```
+
+This will collate all the found misspellings in the source tree and write
+them to `.words`. If `gospel` is run again, it will accept the all words in
+the source tree as correctly spelled as they now appear in the projects
+dictionary. Since hunspell dictionaries are just text files, you can go
+through the list of words to remove properly identified misspellings and
+leave in words that are domain-specific and incorrectly marked.
+
+For example the `.words` file at the root of the `gospel` repo includes
+words that would otherwise be flagged with the default "en_US" spelling
+dictionary and was autogenerated using the command above.
+
+```
+{{.words -}}
+```
+
+This file can also be edited to make use of more advanced hunspell
+features. See [Hunspell Dictionaries](#hunspell-dictionaries) below.
+
+
+## Command Line Options
+
+`gospel` provides a number of command line options. The full list can
+be obtained by running `gospel -h`, but the majority correspond directly
+with `.gospel.conf` options and are described [there](.gospel.conf).
+
+The remaining options are not intended to be persistently stored:
+
+- `-config` — whether to use config file (default true, intended for debugging use).
```

Comment:
if this is only intended for debugging, did you consider an undocumented env var? then the average or new user doesn't have to know about the knob at all.

Reply:
It's intended for user debugging during tuning.